File: src/dataset/go/proverb_Exercism_go_drapala/openAI/ChatGPT4_proverb_v3.go
package proverb

import (
	"fmt"
)

func ProverbMaker(first, second string) string {
	if second == "" {
		return fmt.Sprintf("And all for the want of a %s.", first)
	}
	return fmt.Sprintf("For want of a %s the %s was lost.", first, second)
}

func Proverb(rhyme []string) []string {
	n := len(rhyme)
	if n == 0 {
		return nil
	}

	proverb := make([]string, n)
	for i := 0; i < n-1; i++ {
		proverb[i] = ProverbMaker(rhyme[i], rhyme[i+1])
	}
	proverb[n-1] = ProverbMaker(rhyme[0], "")
	return proverb
}

File: src/dataset/typescript/proverb_Exercism-typescript-shybyte/openAI/ChatGPT4_proverb_v3.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = proverb;
function proverb(...nouns) {
    if (nouns.length === 0)
        return '';
    const result = [];
    for (let i = 0; i < nouns.length - 1; i++) {
        result.push(`For want of a ${nouns[i]} the ${nouns[i + 1]} was lost.`);
    }
    result.push(`And all for the want of a ${nouns[0]}.`);
    return result.join('\n');
}
